coderd/config.go

```diff
@@ -0,0 +1,51 @@
+package coderd
+
+import (
+	"context"
+	"strings"
+
+	"github.com/google/cel-go/common/types"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/v2/coderd/cel"
+	"github.com/coder/coder/v2/codersdk"
+)
+
+// CompileTokenLifetimeExpression parses and compiles the CEL expression for token lifetime evaluation.
+// This should be called during server startup after config is loaded.
+func CompileTokenLifetimeExpression(ctx context.Context, sessionsConfig *codersdk.SessionLifetime, logger slog.Logger) error {
+	expr := strings.TrimSpace(sessionsConfig.MaximumTokenDurationExpression.Value())
+	if expr == "" {
+		logger.Debug(ctx, "role token lifetime expression is empty, using global max for all users")
+		return nil
+	}
+
+	// Create CEL environment with duration function
+	env, err := cel.NewTokenLifetimeEnvironment(cel.EnvironmentOptions{})
+	if err != nil {
+		return xerrors.Errorf("failed to create CEL environment: %w", err)
+	}
+
+	// Compile expression
+	ast, issues := env.Compile(expr)
+	if issues.Err() != nil {
+		return xerrors.Errorf("CEL compilation failed: %w", issues.Err())
+	}
+
+	// Verify the return type is a duration
+	if !ast.OutputType().IsExactType(types.DurationType) {
+		return xerrors.Errorf("CEL expression must return a duration, got %s", ast.OutputType())
+	}
+
+	// Create program
+	program, err := env.Program(ast)
+	if err != nil {
+		return xerrors.Errorf("CEL program creation failed: %w", err)
+	}
+
+	// Cache the compiled program
+	sessionsConfig.SetCompiledMaximumTokenDurationProgram(program)
+	logger.Info(ctx, "successfully compiled role token lifetime expression")
+	return nil
+}
```

Comment:
Should this should be a method of `codersdk.SessionLifetime` instead?

Reply:
yeah, that makes indeed more sense,
Moved it there and added a nice sync.Once to not recompile multiple times.